### PR TITLE
Use libuv thread instead of std::thread to avoid musl issues

### PR DESCRIPTION
**Pull request by gbaraldi**
See https://ariadne.space/2021/06/25/understanding-thread-stack-sizes-and-how-alpine-is-different/ and https://git.alpinelinux.org/aports/commit/?id=4ab6ef38f167f4983008f8b213e5beace8d453f0

### CODE REVIEW

File: Make.inc [hunk 1]
```diff
@@ -1357,6 +1357,11 @@ endif
 JLIBLDFLAGS += -Wl,-z,noexecstack
 endif
 
+#musl has a really small default stack
```
Comment (giordano) [inline Make.inc:1360]: Strictly speaking I think it's only Alpine Linux, but that probably covers vast majority of musl-based systems (and this option should hopefully be good in all other cases)

Comment (gbaraldi) [reply Make.inc:1360]: https://git.musl-libc.org/cgit/musl/commit/?id=7b3348a98c139b4 Seems to a musl thing. Well the flag is a linker thing but it seems musl respects it since 2018

**Comment by vtjnash** [comment]
I thought we avoided using this code path in all cases, since libuv is already aware that musl's defaults are wrong (it usually will result in memory corruption if receiving any signals on any platforms since it is smaller than even a single a page)

File: Make.inc [hunk 2]
```diff
@@ -1357,6 +1357,7 @@ endif
 JLIBLDFLAGS += -Wl,-z,noexecstack
 endif
 
+
```
Comment (giordano) [inline Make.inc:1360]: Now there's an unrelated change to this file.
```suggestion
```
